src/settings.rs

```diff
@@ -36,7 +36,260 @@ pub struct FacetingSettings {
+    /// | `mxbai-embed-large`      | `334M`       | [View model](https://ollama.com/library/mxbai-embed-large)      |
+    /// | `nomic-embed-text`       | `137M`       | [View model](https://ollama.com/library/nomic-embed-text)       |
+    /// | `all-minilm`             | `23M`,`33M`  | [View model](https://ollama.com/library/all-minilm)             |
+    /// | `snowflake-arctic-embed` | varies       | [View model](https://ollama.com/library/snowflake-arctic-embed) |
+    pub model: String,
+    /// Use it to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
+    ///
+    /// if present, `document_template` must be a [Liquid template](https://shopify.github.io/liquid/).
+    /// Use `{{ doc.attribute }}` to access document field values.
+    /// Meilisearch also exposes a `{{ fields }}` array containing one object per document field, which you may access with `{{ field.name }}` and `{{ field.value }}`.
+    ///
+    /// For best results, use short strings indicating the type of document in that index, only include highly relevant document fields, and truncate long fields.
+    /// Example: "A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub document_template: Option<String>,
+}
+
+/// EXPERIMENTAL
+///
+/// # Example
+/// ```
+/// # use std::collections::HashMap;
+/// # use meilisearch_sdk::settings::{GenericRestEmbedderSettings,GenericRestInputType};
+/// use serde_json::Value;
+/// let embedder_setting = GenericRestEmbedderSettings {
+///   url: Some("http://localhost:12345/api/v1/embed".to_string()),
+///   api_key: Some("SOURCE_API_KEY".to_string()),
+///   dimensions: Some(512),
+///   document_template: Some("A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}".to_string()),
+///   input_field: vec!["data".to_string(), "text".to_string()],
+///   input_type: Some(GenericRestInputType::Text),
+///   query: HashMap::from([("model".to_string(), Value::from("MODEL_NAME")), ("dimensions".to_string(), Value::from(512))]),
+///   path_to_embeddings: vec!["data".to_string()],
+///   embedding_object: vec!["embedding".to_string()],
+/// };
+/// # let expected = r#"{"url":"http://localhost:12345/api/v1/embed","apiKey":"SOURCE_API_KEY","dimensions":512,"documentTemplate":"A document titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}","inputField":["data","text"],"inputType":"text","query":{"dimensions":512,"model":"MODEL_NAME"},"pathToEmbeddings":["data"],"embeddingObject":["embedding"]}"#;
+/// # let expected: GenericRestEmbedderSettings = serde_json::from_str(expected).unwrap();
+/// # assert_eq!(embedder_setting, expected);
+/// ```
+#[derive(Serialize, Deserialize, Default, Debug, Clone, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericRestEmbedderSettings {
```

Comment:
Needs to be adapted to account for the breaking changes in `v1.10.0`:

https://github.com/meilisearch/meilisearch/releases/tag/v1.10.0#:~:text=Changing%20the%20parameters%20of%20the%20REST%20API

Reply:
https://github.com/meilisearch/meilisearch-rust/pull/554/commits/73981850e687a87d7d317dbef7b5ad969e77f78b